### PR TITLE
getdp: init at 3.0.4

### PR DESCRIPTION
**Pull request by wucke13**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by wucke13** [comment]
@GrahamcOfBorg build standard-tar

**Comment by wucke13** [comment]
@GrahamcOfBorg build getdp

File: pkgs/applications/science/math/getdp/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  name = "getdp-${version}";
+  version = "3.0.3";
+  src = fetchurl {
+    url = "http://getdp.info/src/getdp-${version}-source.tgz";
+    sha256 = "1ng96vrbcpb6f757mv9sz4pww7h6bhlwwkzxjapgmrjw2avq081z";
+  };
+
+  buildInputs = [ cmake gfortran blas liblapack openmpi ];
+
+  enableParallelBuilding = true;
```
Comment (nlewo) [inline pkgs/applications/science/math/getdp/default.nix]: I think it's default behavior of Cmake builds. In this case, this is unnecessary.

**Review by nlewo — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/getdp/default.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/applications/science/math/getdp/default.nix]: Removed the `enableParallelBuilding = true;`, builds still with multiple cores. Thanks for the feedback!

**Comment by nlewo** [comment]
@GrahamcOfBorg build getdp

**Comment by nlewo** [comment]
@wucke13 Could you either fix the build for failing arch or remove them from supported platform list?

**Comment by wucke13** [comment]
I have absolutely no clue why the build on `aarch64-linux` failed. Is GHC needed for Gfortran? And if yes, does GHC really do not compile on AArch64?
Regarding the Darwin Build: I'm trying to fix it.

**Comment by wucke13** [comment]
@GrahamcOfBorg build getdp

**Comment by nlewo** [comment]
@wucke13 regarding `aarch64`, then just remove it from the `meta.supportedPlatforms` list.
@GrahamcOfBorg build getdp

**Comment by wucke13** [comment]
@nlewo Adjusted the build according to your suggestion. Also updated to the newest stable release.

**Comment by nlewo** [comment]
@GrahamcOfBorg build getdp

**Comment by wucke13** [comment]
@nlewo May it be the case that @GrahamcOfBorg fails to build due to errors unrelated to `getdp`?

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/getdp/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+stdenv.mkDerivation rec {
+  name = "getdp-${version}";
+  version = "3.0.4";
+  src = fetchurl {
+    url = "http://getdp.info/src/getdp-${version}-source.tgz";
+    sha256 = "0v3hg03lzw4hz28hm45hpv0gyydqz0wav7xvb5n0v0jrm47mrspv";
+  };
+
+  buildInputs = [ cmake gfortran blas liblapack openmpi python3 ];
```
[the remark this answers is not in the document]
Comment (wucke13) [reply pkgs/applications/science/math/getdp/default.nix]: Thanks for the feedback, I adjusted the expression accordingly.

**Comment by markuskowa** [comment]
@GrahamcOfBorg build getdp

**Comment by markuskowa** [comment]
Can you please do rebase. That GHC problem might be gone on the actual master.
Did you try to run the program after switching to openblas?

**Comment by wucke13** [comment]
@markuskowa Thanks for the hint about rebasing. Didn't think of that, but of course that might be a good reason for @GrahamcOfBorg to fail. The resulting binary indeed works. I will submit my changes by tomorrow.

**Comment by markuskowa** [comment]
@GrahamcOfBorg build getdp

**Pull request merged**